src/Source.cpp

```diff
@@ -10,16 +11,26 @@ SourcePtr Source::create(List spec) {
   std::string subclass(as<CharacterVector>(spec.attr("class"))[0]);
 
   int skip = as<int>(spec["skip"]);
-  std::string comment = as<std::string>(spec["comment"]);
+  SEXP comment_sexp = spec["comment"];
```

Comment:
Why not `CharacterVector comment_sexp`? That will throw an error if not correct type, and simplifies the for loop that follows.